crates/torii/libp2p/src/server/mod.rs

```diff
@@ -0,0 +1,228 @@
+                    .map(|(peer_id, conn), _| (peer_id, StreamMuxerBox::new(conn))))
+            })
+            .expect("Failed to create WebRTC transport")
+            .with_behaviour(|key| {
+                let message_id_fn = |message: &gossipsub::Message| {
+                    let mut s = DefaultHasher::new();
+                    message.data.hash(&mut s);
+                    gossipsub::MessageId::from(s.finish().to_string())
+                };
+                let gossipsub_config = gossipsub::ConfigBuilder::default()
+                        .heartbeat_interval(Duration::from_secs(10)) // This is set to aid debugging by not cluttering the log space
+                        .validation_mode(gossipsub::ValidationMode::Strict) // This sets the kind of message validation. The default is Strict (enforce message signing)
+                        .message_id_fn(message_id_fn) // content-address messages. No two messages of the same content will be propagated.
+                        .build()
+                        .map_err(|msg| io::Error::new(io::ErrorKind::Other, msg)).unwrap(); // Temporary hack because `build` does not return a proper `std::error::Error`.
+
+                Behaviour {
+                    relay: relay::Behaviour::new(key.public().to_peer_id(), Default::default()),
+                    ping: ping::Behaviour::new(ping::Config::new()),
+                    identify: identify::Behaviour::new(identify::Config::new(
+                        "/TODO/0.0.1".to_string(),
+                        key.public(),
+                    )),
+                    gossipsub: gossipsub::Behaviour::new(
+                        gossipsub::MessageAuthenticity::Signed(key.clone()),
+                        gossipsub_config,
+                    )
+                    .unwrap(),
+                }
+            })?
+            .with_swarm_config(|cfg| cfg.with_idle_connection_timeout(Duration::from_secs(60)))
+            .build();
+
+        // TCP
+        let listen_addr_tcp = Multiaddr::from(Ipv4Addr::UNSPECIFIED).with(Protocol::Tcp(port));
+        swarm.listen_on(listen_addr_tcp.clone())?;
+
+        // UDP QUIC
+        let listen_addr_quic =
+            Multiaddr::from(Ipv4Addr::UNSPECIFIED).with(Protocol::Udp(port)).with(Protocol::QuicV1);
+        swarm.listen_on(listen_addr_quic.clone())?;
+
+        // WebRTC
+        let listen_addr_webrtc = Multiaddr::from(Ipv4Addr::UNSPECIFIED)
+            .with(Protocol::Udp(port_webrtc))
+            .with(Protocol::WebRTCDirect);
+        swarm.listen_on(listen_addr_webrtc.clone())?;
+
+        // Clients will send their messages to the "message" topic
+        // with a room name as the message data.
+        // and we will forward those messages to a specific room - in this case the topic
+        // along with the message data.
+        swarm.behaviour_mut().gossipsub.subscribe(&IdentTopic::new("message")).unwrap();
```

Comment:
This `"message"` topic may be relocated to a common variable for client and server.